src/notification.c

```diff
@@ -282,6 +282,29 @@ void notification_init(notification *n)
         n->body     = n->body     ? n->body     : g_strdup("");
         n->category = n->category ? n->category : g_strdup("");
 
+        /* log */
+        if (settings.log_notifications) {
+            FILE *log_file;
+            log_file = fopen(settings.log_path, "a");
+
+            for(char *c = settings.log_format; *c; c++) {
```

Comment:
Don't reinvent the wheel, we already have `notification_format_message` that does this exact thing and can be easily repurposed (hint: make `format` a parameter).